pkg/controller/queuejob/queuejob_controller_ex.go

```diff
@@ -1107,8 +1095,21 @@ func (qjm *XController) ScheduleNext() {
 				if qjm.serverOption.DynamicPriority {
 					priorityindex = -math.MaxFloat64
 				}
+				//cache.go updatestate method fails resulting in empty resource object
+				//cache upate failure costly, as it will put current AW in backoff queue plus take another dispatch cycle
+				//In worst case the cache update could fail for subsequent dispatche cycles causing test cases to fail or AW never getting dispatched
+				//To avoid non-determinism below code is workaround. this should be issue should be fixed: https://github.com/project-codeflare/multi-cluster-app-dispatcher/issues/550
+				var unallocatedResources = clusterstateapi.EmptyResource()
+				unallocatedResources = qjm.cache.GetUnallocatedResources()
+				for unallocatedResources.IsEmpty() {
+					unallocatedResources.Add(qjm.cache.GetUnallocatedResources())
+					if !unallocatedResources.IsEmpty() {
+						break
+					}
+				}
+
```

Comment:
Risky business. We definitely need a different logic here.

Reply:
Yes agree, we need a new cache design